src/activity_item_handling.cpp

```diff
@@ -1159,12 +1153,566 @@ void activity_on_turn_blueprint_move( player_activity &, player &p )
+    for( map_stack::iterator it = items_at_post.begin(); it != items_at_post.end(); ) {
+        if( it->has_quality( qual ) ) {
+            units::volume volume_allowed = p->volume_capacity() - p->volume_carried();
+            units::mass weight_allowed = p->weight_capacity() - p->weight_carried();
+            if( volume_allowed <= 0_ml || weight_allowed <= 0_gram ) {
+                add_msg( m_info, _( "%s cannot pick up the %s, it's too big." ), p->disp_name(), it->tname() );
+                break;
+            }
+            add_msg( m_info, _( "%s picks up the %s" ), p->disp_name(), it->tname() );
+            found_post = true;
+            p->i_add( *it );
+            // calls map::i_rem() internally, but is more readable in this context imo
+            it = items_at_post.erase( it );
+            break;
+        }
+    }
+    // guess it's gone missing, try again, maybe its nearby. restart activity.
+    if( !found_post ) {
+        p->assign_activity( act_multiple_fetch );
+        return;
+    }
+}
+
+void activity_on_turn_fetch_seeds( player_activity &act, player *p )
+{
+    ( void )act;
+    const activity_id act_multiple_fetch = activity_id( "ACT_FETCH_SEEDS" );
+    const tripoint abspos = g->m.getabs( p->pos() );
+    const zone_manager &mgr = zone_manager::get_manager();
+    const std::unordered_set<tripoint> &src_set = mgr.get_near(
+                zone_type_id( "FARM_PLOT" ), abspos );
+    const std::vector<tripoint> &src_sorted = get_sorted_tiles_by_distance( abspos, src_set );
+    const activity_id act_multiple_farm = activity_id( "ACT_MULTIPLE_FARM" );
+
+    using seed_map_t = std::map<std::string, int>;
+    seed_map_t seed_names;
+    // check how many seeds we may require
+    for( const tripoint &src : src_sorted ) {
+        const zone_data *zone = mgr.get_zone_at( src );
+        if( !zone ) {
+            continue;
+        }
+        tripoint src_loc = g->m.getlocal( src );
+        if( g->m.has_flag_ter_or_furn( "PLANTABLE", src_loc ) && !g->m.has_items( src_loc ) &&
+            !g->m.has_furn( src_loc ) ) {
+            const auto options = dynamic_cast<const plot_options &>( zone->get_options() );
+            const std::string seed = options.get_seed();
+            std::pair<seed_map_t::iterator, bool> has_insert = seed_names.insert( std::make_pair( seed, 1 ) );
+            if( !has_insert.second ) {
+                seed_names[seed]++;
+            }
+        }
+    }
+    // what do we have in our inventory? compared to what we need?
+    std::vector<item *> seed_inv = p->items_with( []( const item & itm ) {
+        return itm.is_seed();
+    } );
+    for( const item *elem : seed_inv ) {
+        std::map<std::string, int>::iterator map_it;
+        map_it = seed_names.find( elem->typeId() );
+        if( map_it != seed_names.end() ) {
+            // if we have some in our inventory already, remove it from the wanted list.
+            seed_names.find( elem->typeId() )->second -= 1;
```

Comment:
```suggestion
            seed_names.find( elem->typeId() )->second -= elem->count();
```
Because seeds are usually counted by charges, `elem` might be a couple of seeds, not just one seed.